x-pack/plugins/task_manager/task_runner.ts

```diff
@@ -36,6 +36,7 @@ export interface TaskRunner {
 interface Updatable {
   update(doc: ConcreteTaskInstance): Promise<ConcreteTaskInstance>;
   remove(id: string): Promise<RemoveResult>;
+  getMaxAttempts(): number;
```

Comment:
This could provide limitations to the domain of this number. Perhaps `[0-INF)` or `null` to indicate it's unlimited retries.

Reply:
Turns out this is not possible in Typescript currently.

Reply:
Looks like it's not possible to specify a range for a number type in Typescript, yet :)